stacks/api/passport/auth.guard.ts

```diff
@@ -15,16 +15,17 @@ export type IAuthGuard = CanActivate & {
   logIn<TRequest extends {logIn: Function} = any>(request: TRequest): Promise<void>;
   handleRequest<TUser = any>(err, user, info, context): TUser;
 };
-export const AuthGuard: (type?: string | string[]) => Type<IAuthGuard> = memoize(createAuthGuard);
+export const AuthGuard: (type?: string) => Type<IAuthGuard> = memoize(createAuthGuard);
 
 const NO_STRATEGY_ERROR = `In order to use "defaultStrategy", please, ensure to import PassportModule in each place where AuthGuard() is being used. Otherwise, passport won't work correctly.`;
 
-function createAuthGuard(type?: string | string[]): Type<CanActivate> {
+function createAuthGuard(type?: string): Type<CanActivate> {
   class MixinAuthGuard<TUser = any> implements CanActivate {
     constructor(@Optional() protected readonly options?: AuthModuleOptions) {
-      this.options = this.options || {};
-      if (!type && !this.options.defaultStrategy) {
-        new Logger("AuthGuard").error(NO_STRATEGY_ERROR);
+      if (Array.isArray(this.options.defaultStrategy)) {
+        throw "Default strategy can not be array.";
+      } else {
+        type = this.options.defaultStrategy;
```

Comment:
Can you also check if the default strategy has provided? Because it has to be provided in case the client does not send a valid scheme. When the client does send an invalid scheme or does not send it at all, we have to fall back to the default strategy and authorize the user against it.